### PR TITLE
examples: multiplex

### LINKED ISSUE #2486: examples: channel/server sharing
Show that a channel can be shared by two stubs and two servicers can share a server.

### PR DESCRIPTION
**Pull request by menghanl**
Show that a channel can be shared by two stubs and two services can share a
server.

fixes #2486

### CODE REVIEW

File: examples/features/multiplex/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Multiplex
+
+A `ClientConn` can be shared by two stubs and two servicers can share a server.
```
Comment (dfawley) [inline examples/features/multiplex/README.md]: "...can share a `Server`."

Or `grpc.ClientConn` and `grpc.Server`?

Comment (dfawley) [inline examples/features/multiplex/README.md]: Add something like: "This example illustrates how to perform both types of sharing."

Comment (menghanl) [reply examples/features/multiplex/README.md]: Done

Comment (menghanl) [reply examples/features/multiplex/README.md]: Done

File: examples/features/multiplex/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Multiplex
+
+A `grpc.ClientConn` can be shared by two stubs and two servicers can share a
```
Comment (dfawley) [inline examples/features/multiplex/README.md]: servicers->services

File: examples/features/multiplex/client/main.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	r, err := c.SayHello(ctx, &hwpb.HelloRequest{Name: name})
+	if err != nil {
+		log.Fatalf("could not greet: %v", err)
+	}
+	fmt.Println("Greeting: ", r.Message)
+}
+
+func callUnaryEcho(client ecpb.EchoClient, req *ecpb.EchoRequest) {
```
Comment (dfawley) [inline examples/features/multiplex/client/main.go]: To be consistent this should accept a string parameter or the other function should accept the request message.

File: examples/features/multiplex/README.md (hunk 2 above)
Comment (menghanl) [reply examples/features/multiplex/README.md]: Done

File: examples/features/multiplex/client/main.go (hunk 3 above)
Comment (menghanl) [reply examples/features/multiplex/client/main.go]: Done

**Pull request merged**